snuba/consumers/consumer.py

```diff
@@ -402,6 +436,9 @@ def close(self) -> None:
                 try:
                     step.close()
                 except Exception as e:
+                    if self.__dead_letter_step:
+                        # TODO: self.__dead_letter_step.submit
+                        pass
```

Comment:
Are you sure that the exception should be caught here and not in poll?
@nikhars  ?

Reply:
This is the right place to catch the exception